src/Evolution/Systems/ScalarWave/BoundaryConditions/ConstraintPreservingSphericalRadiation.hpp

```diff
@@ -0,0 +1,264 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+#include <type_traits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/BoundaryConditions/Type.hpp"
+#include "Evolution/Systems/ScalarWave/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/AnalyticData/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain::Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace ScalarWave::BoundaryConditions {
+namespace detail {
+/// The type of spherical radiation boundary condition to impose
+enum class ConstraintPreservingSphericalRadiationType {
+  /// Impose \f$(\partial_t + \partial_r)\Psi=0\f$
+  Sommerfeld,
+  /// Impose \f$(\partial_t + \partial_r + r^{-1})\Psi=0\f$
+  FirstOrderBaylissTurkel,
+  /// Imposes a second-order Bayliss-Turkel boundary condition
+  SecondOrderBaylissTurkel
+};
+
+ConstraintPreservingSphericalRadiationType
+convert_constraint_preserving_spherical_radiation_type_from_yaml(
+    const Options::Option& options);
+}  // namespace detail
+
+/*!
+ * \brief Constraint-preserving spherical radiation boundary condition that
+ * seeks to avoid ingoing constraint violations and radiation.
+ *
+ * The constraint preserving part of the boundary condition imposes the
+ * following condition on the time derivatives of the characteristic fields:
+ *
+ * \f{align*}{
+ *   d_tw^\Psi&\to d_tw^{\Psi}+\lambda_{\Psi}n^i\mathcal{C}_i, \\
+ *   d_tw^0_i&\to d_tw^{0}_i+\lambda_{0}n^jP^k_i\mathcal{C}_{ik},
+ * \f}
+ *
+ * where
+ *
+ * \f{align*}{
+ * P^k{}_i=\delta^k_i-n^kn_i
+ * \f}
+ *
+ * is the projection operator, and \f$d_t w\f$ is the evolved to characteristic
```

Comment:
1. Clarify which projection.. to the boundary?
2. Is this just a confusing way of saying "the time derivative of the characteristic fields"? Do these operators not commute? If it matters that you're talking about the transformation and not the fields, then I suggest using an upper-case letter or something to help the understanding that this is an operator and not a field.